internal/cloud/cloud_discovery.go

```diff
@@ -34,25 +34,32 @@ type DiscoveryClient struct {
 	logger     logger.LogAdaptor
 	httpClient *rest.HTTPClient
 	token      string
+	baseURL    string
 }
 
 func NewDiscoveryClient(config *cluster.CloudConfig, logger logger.LogAdaptor) *DiscoveryClient {
+	url := config.ExperimentalAPIBaseURL
```

Comment:
You propose updating `baseAPIURL` signature so it takes a `cluster.CloudConfig` instead of adding it to `cluster.CloudConfig`. That's possible but `baseAPIURL` is intended to act more like the `defaultBaseAPIURL`. I'll rename it.